fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/ExternalContentHandlerIT.java

```diff
@@ -67,4 +99,531 @@ public void testRemoteUriContentType() throws Exception {
+            assertEquals("attachment", disposition.getType());
+        }
+    }
+
+    @Test
+    public void testGetExternalDatastream() throws IOException, ParseException {
+        final String id = getRandomUniqueId();
+        final HttpPut put = putObjMethod(id);
+        put.addHeader(LINK, getExternalContentLinkHeader("http://example.com/test", "redirect", "image/jpeg"));
+        assertEquals(CREATED.getStatusCode(), getStatus(put));
+
+        // Configure HEAD request to NOT follow redirects
+        final HttpGet getObjMethod = getObjMethod(id);
+        final RequestConfig.Builder requestConfig = RequestConfig.custom();
+        requestConfig.setRedirectsEnabled(false);
+        getObjMethod.setConfig(requestConfig.build());
+
+        try (final CloseableHttpResponse response = execute(getObjMethod)) {
+            assertEquals(TEMPORARY_REDIRECT.getStatusCode(), response.getStatusLine().getStatusCode());
+            assertEquals("http://example.com/test", getLocation(response));
+            assertEquals("bytes", response.getFirstHeader("Accept-Ranges").getValue());
+            final ContentDisposition disposition =
+                    new ContentDisposition(response.getFirstHeader(CONTENT_DISPOSITION).getValue());
+            assertEquals("attachment", disposition.getType());
+        }
+    }
+
+    @Test
+    public void testHeadExternalDatastreamWithWantDigest() throws IOException, ParseException {
+
+        final String dsId = getRandomUniqueId();
+        createDatastream(dsId, "x", TEST_BINARY_CONTENT);
+
+        final String dsUrl = serverAddress + dsId + "/x";
+
+        final String id = getRandomUniqueId();
+        final HttpPut put = putObjMethod(id);
+        put.addHeader(LINK, getExternalContentLinkHeader(dsUrl, "redirect", "text/plain"));
+        assertEquals(CREATED.getStatusCode(), getStatus(put));
+
+        // Configure HEAD request to NOT follow redirects
+        final HttpHead headObjMethod = headObjMethod(id);
+        headObjMethod.addHeader(WANT_DIGEST, "sha");
+        final RequestConfig.Builder requestConfig = RequestConfig.custom();
+        requestConfig.setRedirectsEnabled(false);
+        headObjMethod.setConfig(requestConfig.build());
+
+        try (final CloseableHttpResponse response = execute(headObjMethod)) {
+            assertEquals(TEMPORARY_REDIRECT.getStatusCode(), response.getStatusLine().getStatusCode());
+            assertTrue(response.getHeaders(DIGEST).length > 0);
+            assertEquals(dsUrl, getLocation(response));
+            final String digesterHeaderValue = response.getHeaders(DIGEST)[0].getValue();
+            assertTrue("Fixity Checksum doesn't match",
+                    digesterHeaderValue.equals(TEST_SHA_DIGEST_HEADER_VALUE));
+        }
+    }
+
+    @Test
+    public void testHeadExternalDatastreamWithWantDigestMultiple() throws IOException, ParseException {
```

Comment:
Should this test and the one below it (line 388) be combined and renamed to identify the "redirect". Sort of like the tests [`testExternalDatastreamCopyWithWantDigest()`](https://github.com/fcrepo4/fcrepo4/pull/1386/files#diff-ddd58a417a8d65f87d229b3c7e54faf0R180)

Reply:
I have merged the redirect want-digest tests together